doc/evp_skey.md

```diff
@@ -0,0 +1,90 @@
+EVP_SKEY Design
+===============
+
+We currently presume that symmetric keys are represnted as octet buffer. It may
+be not relevant for the situation of hardware-backed providers, such as
+PKCS#11 or TPM.
+
+This is a design of using opaque usage in symmetric encryption.
+
+
+Creation
+--------
+
+```
+EVP_SKEY * EVP_SKEY_new(void)
+```
+creates an empty `EVP_SKEY` object.
+
+API to set parameters
+---------------------
+
+```
+int EVP_SKEY_fromdata(EVP_CIPHER_CTX *ctx, EVP_SKEY **pskey, OSSL_PARAM params[]);
+int EVP_SKEY_assign(EVP_CIPHER_CTX *ctx, EVP_SKEY *pskey, OSSL_PARAM params[]);
+```
+
+API to derive a symmetric key
+-----------------------------
+
+```
+int EVP_PKEY_derive_SKEY(EVP_PKEY_CTX *ctx, EVP_SKEY *pskey);
+```
+similar to `EVP_PKEY_derive`
```

Comment:
What about the EVP_KDF APIs?

Reply:
I'd go further and ask if we want to update the legacy PKEY access to KDFs?
I.e. only support SKEY via the EVP_KDF calls.